runtime_docker.go

```diff
@@ -75,11 +76,13 @@ func (r *RuntimeDocker) Start(ctx context.Context) (RuntimeContext, error) {
 		return RuntimeContext{}, errors.Wrap(err, "cannot create Docker client using environment variables")
 	}
 
-	out, err := c.ImagePull(ctx, r.Image, types.ImagePullOptions{})
-	if err != nil {
-		return RuntimeContext{}, errors.Wrapf(err, "cannot pull Docker image %q", r.Image)
+	if found := strings.Contains(r.Image, "/"); found {
```

Comment:
I like this idea, but I could imagine you might want to use a local image that you'd tagged with a `/` but not yet pushed.

I can't think of a foolproof way to automatically detect when something needs to be pulled. What do you think about making this a specific configuration knob. We could support an annotation like `xrender.crossplane.io/runtime-docker-pullpolicy` that would set a corresponding `PullPolicy` option in the `RuntimeDocker` struct.

Reply:
yes, I agree a flag similar to `docker run --pull` could be useful:
```shell
--pull string                    Pull image before running ("always", "missing", "never") (default "missing")
```

Looks like docker is [handling](https://github.com/docker/cli/blob/cfe00daaaa806d7a05a5fce1c0b3c6b051f1da6b/cli/command/container/create.go#L255) it by:
- pulling and then creating the container if set to `always`
- trying to run the container, catching the error and if it's a not found, they either pull it and retry creating the container  if set to `missing`, or they just error out if set to `never`

We should do the same.